sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -156,6 +156,11 @@ abstract class ColumnarOutputWriter(context: TaskAttemptContext,
     }
   }
 
+  /** Apply any necessary casts before writing batch out */
+  def transform(cb: ColumnarBatch): ColumnarBatch = {
+    cb
+  }
+
   private[this] def writeBatchWithRetry(batch: ColumnarBatch): Long = {
```

Comment:
so `writeBatchWithRetry` and `writeBatchNoRetry` are being called from `writeBatch`. The contract is that `writeBatch` should close the batch.